### PR TITLE
dadjoke trigger refactor

### PR DESCRIPTION
**Pull request by parsec**
This time I actually wrote it to receive JSON data and stuff. This time I did it the smahrt way. Yup. I think.

### CODE REVIEW

**Comment by parsec** [comment]
so uh, yeah, I think I made it work right now

File: dadjokes.go [hunk 1]
```diff
@@ -16,6 +17,12 @@ func dadjoke() string {
 
 func httpRequest() string {
 
+	type dadJoke struct {
```
Comment (bbriggs) [inline dadjokes.go:20]: Types belong at global scope

File: dadjokes.go [hunk 2]
```diff
@@ -49,9 +59,18 @@ func httpRequest() string {
...
 	r, err := ioutil.ReadAll(resp.Body)
+
 	if err != nil {
 		sublogger.Warn().Err(err).Msg("ioutil.ReadAll error")
 		return fmt.Sprintf("An error occured: %v", err)
 	}
-	return string(r)
+
+	dadJokeResp := new(dadJoke)
+	unmarshalErr := json.Unmarshal([]byte(r), &dadJokeResp)
```
Comment (bbriggs) [inline dadjokes.go:69]: Add an unmarshal method to the dadjoke type

**Review by m-242 — APPROVED** [review]
The tests might break when we get automated testing, as we are making the API calls for real, but it's fine for now, I guess.

**Pull request merged**